### PR TITLE
MWPW-152411: MAS technical documentation

### PR DESCRIPTION
**Pull request by yesil**
Introduces md file based technical documentation generation for M@S.

Contains:
- https://mwpw-152411--milo--yesil.hlx.page/libs/features/mas/docs/mas.html
- https://mwpw-152411--milo--yesil.hlx.page/libs/features/mas/docs/mas.js.html
- https://mwpw-152411--milo--yesil.hlx.page/libs/features/mas/docs/checkout-link.html

Resolves: [MWPW-152411](https://jira.corp.adobe.com/browse/MWPW-152411)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://mwpw-152411--milo--yesil.hlx.page/?martech=off

### CODE REVIEW

File: libs/features/mas/commerce/checkout-link.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+#### Demo<br>
+
+This is a functional **buy now** button: <a href="#" is="checkout-link" data-wcs-osi="A1xn6EL4pK93bWjM8flffQpfEL-bnvtoQKQAvkx574M">Buy now</a>
+
+
+## Attributes {#attributes}
+
+| Attribute                    | Description                                                                                     | Default Value | Required |
+|------------------------------|-------------------------------------------------------------------------------------------------|---------------|----------|
+| `data-wcs-osi`               | Offer Selector ID, can be multiple, seperated by comma           |               |   `true`       |
```
Comment (mirafedas) [inline libs/features/mas/commerce/checkout-link.md:40]: seperated > separated

File: libs/features/mas/commerce/checkout-link.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+| `data-checkout-workflow`     | Target checkout workflow to for the generation of checkout urls     | UCv3              |         `false` |
```
Comment (mirafedas) [inline libs/features/mas/commerce/checkout-link.md:41]: 'to for' > probably just 'for'?

File: libs/features/mas/commerce/checkout-link.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+| `data-checkout-workflow-step`| [workflow step](https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=businessservices&title=UCv3+Link+Creation+Guide#UCv3LinkCreationGuide-RegularWorkflow) to land on on the unified checkout page|     email          |     `false`     |
```
Comment (mirafedas) [inline libs/features/mas/commerce/checkout-link.md:42]: 'on on the' > 'on the'

File: libs/features/mas/commerce/checkout-link.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+| `requestUpdate(true\|false)` |  causes a re-render using the actual options      |
+
+
+## Events {#events}
+
+| Event     | Description                       |
+|-----------|-----------------------------------|
+| `wcms:placeholder:pending` | fires when checkout link starts loading    |
+| `wcms:placeholder:resolved`| fires when the offer is successfully  resolved  |
+| `wcms:placeholder:failed`  | fires when the offer is could not be found or fetched     |
```
Comment (mirafedas) [inline libs/features/mas/commerce/checkout-link.md:146]: 'is could' > 'could'?

**Review by npeltier — APPROVED** [review]

File: libs/features/mas/commerce/checkout-link.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+| `click`   | native click event on the `a`  element      |
+
+
+<br>
+
+For each events except `click`, the following css classes are toggled on the element: `placeholder-pending`, `placeholder-resolved`, `placeholder-failed`.
```
Comment (mirafedas) [inline libs/features/mas/commerce/checkout-link.md:152]: 'For each events' > 'For each event'

**Review by mirafedas — CHANGES_REQUESTED** [review]

**Review by mirafedas — APPROVED** [review]

**Comment by npeltier** [comment]
@yesil shouldn't we keep mas.md instead of mas.js.md, or move checkout-link.md to checkout-link.js.md, to me js extension is not really bringing any value here, it's even rather confusing

**Comment by yesil** [comment]
@npeltier  mas.md is the general introduction to project MAS  whereas mas.js.md is about the library we ship for the enablement.

**Pull request merged**